### PR TITLE
Improvements to EPT provider file filtering, more point cloud handling in provider registry

### PR DESCRIPTION
**Pull request by nyalldawson**
Refs discussion in #39780

### CODE REVIEW

**Comment by nyalldawson** [comment]
Ping @PeterPetrik

File: src/core/qgsfileutils.cpp [hunk 1]
```diff
@@ -56,6 +56,38 @@ QStringList QgsFileUtils::extensionsFromFilter( const QString &filter )
   return extensions;
 }
 
+bool QgsFileUtils::fileMatchesFilter( const QString &fileName, const QString &filter )
```
Comment (PeterPetrik) [inline src/core/qgsfileutils.cpp:59]: wondeering how this behaves for filters or filenames without extension (e.g. bash scripts or README)

File: src/core/qgsprovidermetadata.h [hunk 2]
```diff
@@ -215,10 +216,11 @@ class CORE_EXPORT QgsProviderMetadata : public QObject
...
     {
-      FilterVector = 1,
-      FilterRaster,
-      FilterMesh,
-      FilterMeshDataset
+      FilterVector = 1, //!< Vector layers
+      FilterRaster, //!< Raster layers
+      FilterMesh, //!< Mesh layers
+      FilterMeshDataset, //!< Mesh datasets
+      FilterPointCloud, //!< Point clouds
```
Comment (PeterPetrik) [inline src/core/qgsprovidermetadata.h:223]: since 3.18?

**Review by PeterPetrik — APPROVED** [review]

File: src/core/qgsfileutils.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/qgsfileutils.cpp:59]: I did originally have a test in place for this, but had to remove it as the fallback I've used for qt < 5.12 doesn't support it. Given we'll be forcing 5.12+ in the near future I'll resurrect this test at that time

**Pull request merged**